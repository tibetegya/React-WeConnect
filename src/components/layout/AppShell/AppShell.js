import React from 'react'
import PropTypes from 'prop-types'

import Navbar from '../Navbar'
import Footer from '../Footer'

const AppShell = props => {
  return (
    <div data-testid="app-shell">
      <Navbar />
      AppShell
      <Footer />
    </div>
  )
}

AppShell.propTypes = {

}

export default AppShell
