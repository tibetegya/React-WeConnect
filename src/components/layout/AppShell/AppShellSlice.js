import { createSlice } from '@reduxjs/toolkit'

export const AppShellSlice = createSlice({
  name: 'appShell',
  initialState: { default: true },
  reducers: {
    changeDefault: (state) => {
      state.default = !state.default
    }
  }
})