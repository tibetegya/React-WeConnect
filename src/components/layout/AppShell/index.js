export { default as AppShell } from './AppShell';
export { AppShellSlice } from './AppShellSlice';
