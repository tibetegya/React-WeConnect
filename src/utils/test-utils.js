import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider as ReduxProvider } from 'react-redux';
import store from '../redux/store';
import { defaultTheme } from './themes';

const AllTheProviders = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={defaultTheme}>
      {children}
    </ThemeProvider>
  </ReduxProvider>
);

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
