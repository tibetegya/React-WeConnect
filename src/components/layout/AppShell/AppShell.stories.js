import React from 'react';
import { withA11y } from '@storybook/addon-a11y';
import AppShell from './AppShell'

export default {
  title: 'AppShell',
  component: AppShell,
  decorators: [ withA11y ]
}

export const AppShellStorys = () => (<AppShell />)