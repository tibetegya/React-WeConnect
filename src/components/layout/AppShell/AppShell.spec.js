import React from 'react'
import { render } from 'test-utils'
import AppShell from './AppShell'

describe('AppShell Component tests', () => {
  it('should render corectly', () => {
    const { getByTestId } = render(<AppShell />)
    expect(getByTestId('app-shell')).toMatchSnapshot()
  })
})