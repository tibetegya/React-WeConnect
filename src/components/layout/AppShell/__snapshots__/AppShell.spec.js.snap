// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppShell Component tests should render corectly 1`] = `
<div
  data-testid="app-shell"
>
  <div>
    Navbar
  </div>
  AppShell
  <div>
    Footer
  </div>
</div>
`;
